From 7a8c67b7af13ef0af2fd0b8242414e49fb801e51 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 3 Jun 2010 10:09:46 -0300
Subject: [PATCH 15/20] vnc: send desktopresize event as reply to set encodings

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1275559789-2280-3-git-send-email-kraxel@redhat.com>
Patchwork-id: 9694
O-Subject: [RHEL-6 kvm PATCH 2/5] vnc: send desktopresize event as reply to set
	encodings
Bugzilla: 590070
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

In case the desktop did resize while the vnc connection setup was still
in progress the client isn't informed about it.  Send a desktop resize
event as soon as the client told us it can handle deskop resize via set
encodings message to make sure the client us up to date.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>

bugzilla: #590070 -- QEMU misses DESKTOP-RESIZE event if it is triggered
                     during client connection initialization
upstream: 6356e472512c55a5dc3aaf12ce12a6eb916dd045
---
 vnc.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 vnc.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/vnc.c b/vnc.c
index 68c1ff3..9be6ac7 100644
--- a/vnc.c
+++ b/vnc.c
@@ -1832,6 +1832,7 @@ static void set_encodings(VncState *vs, int32_t *encodings, size_t n_encodings)
         }
     }
 
+    vnc_desktop_resize(vs);
     check_pointer_type_change(vs, kbd_mouse_is_absolute());
 }
 
-- 
1.7.0.3

